experimental/lib/Transforms/Speculation/HandshakeSpeculation.cpp

```diff
@@ -337,21 +340,42 @@ std::optional<Value> findControlInputToBB(Operation *op) {
   handshake::FuncOp funcOp = op->getParentOfType<handshake::FuncOp>();
   assert(funcOp && "op should have parent function");
 
-  // Find input arcs to BBs in the IR
-  BBtoArcsMap bbToPredecessorArcs = getBBPredecessorArcs(funcOp);
-  unsigned bb = getLogicBB(op).value();
+  std::optional<unsigned> targetBB = getLogicBB(op);
+  if (!targetBB) {
+    op->emitError("Operation does not have a BB.");
+    return {};
+  }
+
+  // We use the control token, which is an input to the control branch
+  // as the enable signal for the speculator.
+  mlir::Value ctrlSignal;
+  bool isControlBranchFound = false;
+  for (auto branchOp : funcOp.getOps<handshake::ConditionalBranchOp>()) {
```

Comment:
add comment explaining the loop, use more whitespace to separate out distinct areas of code

Reply:
"Find the control branch we want to speculate on. 
To find: Iterate over every branch, looking for 1) same bb as speculator, and 2) is a control branch"